packages/blade/src/components/ActionList/ActionList.tsx

```diff
@@ -1,115 +1,136 @@
 import React from 'react';
+import styled from 'styled-components';
+import { componentIds } from './componentIds';
 import Box from '~components/Box';
+import type { OptionsType } from '~components/Dropdown/useDropdown';
 import { useDropdown } from '~components/Dropdown/useDropdown';
-import { Text } from '~components/Typography';
-import { getPlatformType, makeAccessible } from '~utils';
-
-type ActionListItemProps = {
-  title: string;
-  value: string;
-  href?: string;
-  /**
-   * Internally passed from ActionList. No need to pass it explicitly
-   */
-  index?: number;
+import {
+  getComponentId,
+  isReactNative,
+  isValidAllowedChildren,
+  makeAccessible,
+  makeSize,
+} from '~utils';
+
+type ActionListProps = {
+  children: React.ReactNode[];
+  surfaceLevel?: 2 | 3;
 };
-const ActionListItem = (props: ActionListItemProps): JSX.Element => {
+
+const StyledActionList = styled(Box)<{ surfaceLevel: ActionListProps['surfaceLevel'] }>(
+  ({ theme, surfaceLevel = 2 }) => {
+    const offsetX = theme.shadows.offsetX.level[1];
+    const offsetY = theme.shadows.offsetY.level[2];
+    const blur = theme.shadows.blurRadius.level[2];
+    const shadowColor = theme.shadows.color.level[1];
+
+    const elevation200 = `${makeSize(offsetX)} ${makeSize(offsetY)} ${makeSize(
+      blur,
+    )} 0px ${shadowColor}`;
+    const backgroundColor = theme.colors.surface.background[`level${surfaceLevel}`].lowContrast;
+
+    return {
+      backgroundColor,
+      borderWidth: theme.border.width.thin,
+      borderColor: theme.colors.surface.border.normal.lowContrast,
+      borderRadius: makeSize(theme.border.radius.medium),
+      padding: makeSize(theme.spacing[3]),
+      boxShadow: isReactNative() ? undefined : elevation200,
+    };
+  },
+);
+
+const ActionList = ({ children, surfaceLevel = 2 }: ActionListProps): JSX.Element => {
   const {
-    activeIndex,
+    setOptions,
+    actionListRef,
+    selectionType,
     dropdownBaseId,
-    onOptionClick,
-    selectedIndices,
-    setShouldIgnoreBlur,
-    selectInputRef,
+    setSelectedIndices,
+    optionsRecalculateToggle,
   } = useDropdown();
+  const actionListOptions: OptionsType = [];
 
-  const platformType = getPlatformType();
-  const renderOnWebAs = props.href ? 'a' : 'button';
-  const isReactNative = platformType === 'react-native';
+  const defaultSelectedIndices: number[] = [];
```

Comment:
Do we not need to keep these two values in React state?

Reply:
We have `selectedIndiced` and `options` state on top level in the context. We just add these local variables to that context so don't need to maintain separate state for these in particular

Reply:
Should be fine, just I hope there will be no stale values :p